lib/ruby_lsp/document.rb

```diff
@@ -89,20 +92,17 @@ def parse
       return if @unparsed_edits.empty?
 
       @unparsed_edits.clear
-      @tree = SyntaxTree.parse(@source)
-      @syntax_error = false
-    rescue SyntaxTree::Parser::ParseError
-      @syntax_error = true
+      @parse_result = YARP.parse(@source)
     end
 
     sig { returns(T::Boolean) }
     def syntax_error?
-      @syntax_error
+      @parse_result.failure?
     end
 
     sig { returns(T::Boolean) }
     def parsed?
-      !@tree.nil?
+      !@parse_result.value.nil?
```

Comment:
@vinistock you had removed this but I re-added since some other classes still rely on it, but changed the behaviour. We can re-visit later confirm if this is the right approach, I'm just aiming to get the tests green first.